src/command/handlers/http/undici.ts

```diff
@@ -0,0 +1,469 @@
+				result.tls = {
+					authorized: tlsSocket.authorized,
+					protocol: tlsSocket.getProtocol(),
+					cipherName: tlsSocket.getCipher()?.name,
+					...(tlsSocket.authorizationError ? { error: tlsSocket.authorizationError } : {}),
+					createdAt: cert.valid_from ? (new Date(cert.valid_from)).toISOString() : null,
+					expiresAt: cert.valid_to ? (new Date(cert.valid_to)).toISOString() : null,
+					issuer: {
+						...(cert.issuer?.C ? { C: cert.issuer.C } : {}),
+						...(cert.issuer?.O ? { O: cert.issuer.O } : {}),
+						...(cert.issuer?.CN ? { CN: cert.issuer.CN } : {}),
+					},
+					subject: {
+						...(cert.subject?.CN ? { CN: cert.subject.CN } : {}),
+						...(cert.subjectaltname ? { alt: cert.subjectaltname } : {}),
+					},
+					keyType: cert.asn1Curve || cert.nistCurve ? 'EC' : cert.modulus || cert.exponent ? 'RSA' : null,
+					keyBits: cert.bits || null,
+					serialNumber: cert.serialNumber?.match(/.{2}/g)?.join(':') ?? null,
+					fingerprint256: cert.fingerprint256,
+					publicKey: cert.pubkey?.toString('hex').toUpperCase().match(/.{2}/g)?.join(':') ?? null,
+				};
+
+				callback(null, tlsSocket);
+			});
+		});
+
+		return tcpSocket;
+	};
+}
+
+export class Test {
+	private readonly REQUEST_TIMEOUT: number = 10_000;
+	private readonly DOWNLOAD_LIMIT: number = 10_000;
+	private readonly url: URL;
+	private readonly port: number;
+	private readonly isHttps: boolean;
+	private undiciClient!: Client;
+	private readonly result: Omit<Result, 'timings'>;
+	private readonly timings: Timings = {
+		start: null,
+		total: null,
+		dns: null,
+		tcp: null,
+		tls: null,
+		firstByte: null,
+		download: null,
+	};
+
+	private resolve!: (value: unknown) => void;
+	private timeoutTimer: NodeJS.Timeout | null = null;
+	private decompressor: Decompressor | null = null;
+	private decompressorHasData = false;
+	private done = false;
+
+	constructor (
+		private readonly options: HttpOptions,
+		private readonly buffer: ProgressBuffer,
+	) {
+		this.result = this.getInitialResult();
+		this.port = options.port ? options.port : (options.protocol === 'HTTP' ? 80 : 443);
+		this.isHttps = options.protocol !== 'HTTP';
+		this.url = new URL(this.urlBuilder());
+	}
+
+	public async run () {
+		const promise = new Promise((resolve) => { this.resolve = resolve; });
+		const dnsResolver = callbackify(dnsLookup(this.options.resolver), true);
+		const allowH2 = this.options.protocol === 'HTTP2';
+		const connector = getConnector(this.options, this.port, this.isHttps, dnsResolver, this.result, this.timings);
+		this.undiciClient = new Client(this.url.origin, { connect: connector, allowH2 });
+		this.timeoutTimer = setTimeout(() => this.handleError('Request timeout.'), this.REQUEST_TIMEOUT);
+
+		this.undiciClient.dispatch({
+			path: this.url.pathname + this.url.search,
+			method: this.options.request.method as Dispatcher.HttpMethod,
+			headers: lowerCaseKeys({
+				'Accept-Encoding': `gzip, deflate, br${createZstdDecompress ? ', zstd' : ''}`,
```

Comment:
This is because of different node versions, right? I think it's better to remove `zstd` for now to avoid hard-to-understand differences in behavior. Got didn't support it either.